pkg/server/datastore/sqlstore/postgres.go

```diff
@@ -13,9 +16,42 @@ import (
 type postgresDB struct{}
 
 func (p postgresDB) connect(cfg *configuration, isReadOnly bool) (db *gorm.DB, version string, supportsCTE bool, err error) {
-	db, err = gorm.Open("postgres", getConnectionString(cfg, isReadOnly))
-	if err != nil {
-		return nil, "", false, sqlError.Wrap(err)
+	if cfg.databaseTypeConfig == nil {
+		return nil, "", false, errors.New("missing datastore configuration")
+	}
+
+	connString := getConnectionString(cfg, isReadOnly)
+	var errOpen error
+	switch {
+	case cfg.databaseTypeConfig.AWSPostgres != nil:
+		c, err := pgx.ParseConfig(connString)
+		if err != nil {
+			return nil, "", false, err
+		}
+		if c.Password != "" {
+			return nil, "", false, errors.New("invalid postgres config: no password should be provided when using IAM authentication")
```

Comment:
```suggestion
			return nil, "", false, errors.New("invalid postgres configuration: password should not be set when using IAM authentication.")
```

Reply:
Thanks for the suggestion. Note that error strings should not end with punctuation, so I'll not end it with ".".